compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -1964,8 +1964,7 @@ public void visit(BLangSimpleVarRef bLangSimpleVarRef) {
             resolvedSymbol.closure = true;
             // When there's a type guard, there can be a enclSymbol before type narrowing.
             // So, we have to mark that as a closure as well.
-            BSymbol enclSymbol = symResolver.lookupClosureVarSymbol(env.enclEnv,
-                    Names.fromString(identifier), SymTag.VARIABLE);
+            BSymbol enclSymbol = symResolver.lookupClosureVarSymbol(env.enclEnv, symbol, Names.fromString(identifier));
```

Comment:
Here also, instead of using `Names.fromString(identifier)`, you can simply use `symbol.name`